[RaspberryPi] Add start and end processing of pet mode.

erk5 opened the pull request:
#16 対応

ペットモードの開始・終了処理を追加しました。

開始条件
- アプリ起動時
- UIの開始クリック時

終了条件
- UIの終了クリック時

終了条件はXX分たったらとかも検討。

erk5 committed:
[RaspberryPi] Add start and end processing of pet mode.

erk5 commented:
@mahya8585 
確認お願いします！

mahya8585 commented on raspberryPi/cocorobo-pet/src/main/java/javajo/config/Constants.java:
```diff
@@ -0,0 +1,18 @@
+		return petMode;
+	}
+
+	public static void setPetMode(boolean petMode) {
```
[nits] lonbok使うのであれば@data付けてsetteer/getter自動生成でもよいのでは

mahya8585 replied:
(nitsなので直す必要ありません）

erk5 replied:
(question) Lombokってstaticでも使えましたっけ？

mahya8585 commented on raspberryPi/cocorobo-pet/src/main/java/javajo/controller/PetController.java:
```diff
@@ -0,0 +1,119 @@
+	 *
+	 * @return メッセージ
+	 */
+	@PutMapping(value = "/pet-mode/start", produces = MediaType.APPLICATION_JSON_VALUE)
```
[question]このAPIはAndroidから呼ばれる認識であってます？

erk5 replied:
Raspberry PiからCocoroboControllerを呼ぶメソッドなのですが、
ペットモードの開始がアプリ起動時だけなのでもう一度Startできるように作ってます。
今はRaspberry Piに作った間に合わせのClient UIから呼ばれます。

mahya8585 commented on raspberryPi/cocorobo-pet/src/main/resources/static/index.html:
```diff
@@ -0,0 +1,43 @@
+	<meta charset="UTF-8">
+	<title>Top</title>
+	<script src="//ajax.googleapis.com/ajax/libs/jquery/3.1.0/jquery.min.js"></script>
+	<script>
```
[IMO]時間に余裕があったらJSファイルにわけよう

mahya8585 commented on raspberryPi/cocorobo-pet/src/main/resources/static/index.html:
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
+<html lang="en">
```
[should] ja

mahya8585 commented on raspberryPi/cocorobo-pet/src/main/resources/static/index.html:
```diff
@@ -0,0 +1,43 @@
+			$('button#start').on('click', function() {
+				$.ajax({
+					type:"put",
+					url:"http://localhost:8080/api/pet-mode/start",
```
[IMO]Azure乗せたときにドメインが変わるので変更する。26行目でも同じドメインを利用するので、余裕があればドメインを変数化して一か所で管理しておきたい（変更漏れ防止）

erk5 replied:
Raspberry PiのRest API呼ぶのでドメインはこのままで大丈夫です。
変数化は余裕があったら直します。

mahya8585 commented:
微々たるコメントだけつけておきました。
内容確認してもらった後、セルフマージしてもらってOKです

erk5 committed:
[RaspberryPi] Code revie.
https://github.com/java-women/cocorobo-pet/pull/53/files/703579e41108fa8969001a1f59c2ab4389548dfa#r80165511

erk5 commented:
index.htmlのlangだけ直してマージしました。
